### ISSUE TITLE
A new dictionary for cybersecurity terms?

### ISSUE THREAD

**Issue by arkid15r** [streetsidesoftware/cspell-dicts#3235]
It seems it would be a good idea to consolidate IT security related terms under a separate dictionary. The dictionary name suggestions:
  - cybersecurity
  - it-security
  - information-security
  - infosec
  - security?

The example terms would be:
  - passcode
  - phising
  - smishing
  - vishing

I guess the closest existing dictionary would be `software-terms` otherwise (not sure, maybe a separate section in this dictionary?)

Essentially, I'm looking for the structure guidance for the example terms.

Thank you!

**Comment by ccoVeille** [comment]
I would say it's the kind of things to add to software terms

The issue I see with splitting dictionary is that people have then to choose between many dictionary. That's why I think it's a better things to consider adding words with tags, and then consider building dictionary on demand from user needs

It's something I plan to consider with my personal project https://github.com/ccoveille/jargon

**Comment by Jason3S** [comment]
@arkid15r,

## **tldr;**

Please create new source files under `software-terms/src` and add them to the `cspell-tools.config.yaml`.

```yaml
  - name: 'softwareTerms'
    sources:
      - filename: src/coding-terms.txt
        split: true
      - filename: src/software-terms.txt
        split: true
      - filename: src/software-tools.txt
        split: true
      # --------> Add new files here. <----------
    format: plaintext
    targetDirectory: './dict'
    generateNonStrict: false
    compress: false
```

## Explanation

What you bring up is the challenge that has evolved as the spell checker has become more popular and the word lists have grown.

The approach I started taking with `software-terms` is to split it out into separate source files while still compiling them into a single dictionary. This allows for future creation of new dictionaries while still providing the benefit now.

Medical and scientific terms are examples of large domain specific sets of term set that warranted their own dictionaries. The vast majority of CSpell users won't need the terms and the ones that do are willing to install the dictionary separately.

As far as performance. The spell checker works better with a few large dictionaries as opposed to many small dictionaries. The cost to look up a word is `len_of_word * num_dictionaries`. The size of the dictionary (beyond loading time) has no impact upon the lookup time. (Note: size does have an impact upon generating suggestions, but that happens several orders of magnitude less frequently.)

**Comment by arkid15r** [comment]
> @arkid15r,
> 
> ## **tldr;**
> Please create new source files under `software-terms/src` and add them to the `cspell-tools.config.yaml`.
> 
> ```yaml
>   - name: 'softwareTerms'
>     sources:
>       - filename: src/coding-terms.txt
>         split: true
>       - filename: src/software-terms.txt
>         split: true
>       - filename: src/software-tools.txt
>         split: true
>       # --------> Add new files here. <----------
>     format: plaintext
>     targetDirectory: './dict'
>     generateNonStrict: false
>     compress: false
> ```

Got it, I'll add it as a part of `software-terms` dictionary.

> As far as performance. The spell checker works better with a few large dictionaries as opposed to many small dictionaries. The cost to look up a word is `len_of_word * num_dictionaries`. The size of the dictionary (beyond loading time) has no impact upon the lookup time. (Note: size does have an impact upon generating suggestions, but that happens several orders of magnitude less frequently.)

Thanks for mentioning that. Now I have better understanding of generally preferred options.
